src/superqt/switch/_toggle_switch.py

```diff
@@ -28,67 +30,74 @@ def __init__(self, parent: QtW.QWidget | None = ...) -> None: ...
     @overload
     def __init__(self, text: str | None, parent: QtW.QWidget | None = ...) -> None: ...
 
-    def __init__(self, text=None, parent=None):
+    def __init__(  # type: ignore [misc]  # overload
+        self, text: str | None = None, parent: QtW.QWidget | None = None
+    ) -> None:
         if isinstance(text, QtW.QWidget):
             if parent is not None:
                 raise TypeError("No overload of QToggleSwitch matches the arguments")
             parent = text
             text = None
+        self._on_color = QtGui.QColor("#4D79C7")
+        self._off_color = QtGui.QColor("#909090")
+        self._handle_color = QtGui.QColor("#d5d5d5")
+        self._offset_value = 8.0
         super().__init__(parent)
-        self._style_option = self.StyleOption()  # the default style option
```

Comment:
it's uncommon to have a "persistent" style option sitting around and holding state.  instead, the widget holds the state and transfers it to a *new* style option during initStyleOption

Reply:
I was thinking that it's better to have a style option inside QToggleSwitch and always copy it in initStyleOption. If we make other options such as `switch_width` overridable, QToggleSwitch will need more and more private attributes.

Reply:
Yes, but that is just the qt pattern. (It’s how all the c code is written as well)… so even if it requires a few more private attributes added to switch itself, it’s a bit less surprising than a long lived styleoption for anyone familiar with qt internals.    Basically style options aren’t meant to maintain state.  They are intended as ephemeral configs to be created and passed to painters for drawing.

(it’s a minor difference, so if you feel strongly about it, it’s fine…)

Reply:
I actually tried to read qtbase but the mysterious `d` attribute appears everywhere and I gave up ...
Ok, in that case, let's go with the qt-internal style 😃

Reply:
Ah yes the mysterious d attribute :). Each widget generally has a private class associated with it (like QCheckBoxPrivate)and it tends to store all the state.  The convention is to call it d think